### ISSUE TITLE
Model CleanUp spits an exception when trying to save a configuration

### ISSUE THREAD

**Issue by DuncanLithgow** [DuncanLithgow/EnTeeEeToolsIssues#4]
I've created a new Configuration. Make all the settings. Click on save all and the tool crashes.

It showed a star by the name before I click Save All (presumably to indicate unsaved changes). That star is gone afterwards. If I click 'Save All' again after this exception then it doesn't complain.

![1elyjaf3bs](https://user-images.githubusercontent.com/3223028/51465327-ff164f80-1d67-11e9-8fc9-1b637a94e806.png)

**Comment by eirannejad** [comment]
Seems like file is in Read-only mode and `Document.Save` fails

**Comment by DuncanLithgow** [comment]
@eirannejad yes. This is a commercial product I'm trying to communicate with about bugs. As I add more bugs and RFEs I hope the level of communication will increase.
